### PR TITLE
fix (api): shared infra group id reference / application permissions

### PR DESCRIPTION
**Pull request by fsamin**

### CODE REVIEW

File: engine/api/application/application_group.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+func AddGroup(db gorp.SqlExecutor, proj *sdk.Project, a *sdk.Application, groupPermission ...sdk.GroupPermission) error {
+	for i := range groupPermission {
+		gp := &groupPermission[i]
+		g := &gp.Group
+		if g.ID == 0 {
+			var errG error
+			g, errG = group.LoadGroup(db, g.Name)
+			if errG != nil {
+				log.Warning("AddGroup: Cannot find %s: %s\n", g.Name, errG)
+				return sdk.ErrGroupNotFound
```
Comment (sguiheux) [inline engine/api/application/application_group.go]: return err.  =>   LoadGroup returns ErrGroupNotFound

File: engine/api/application/application_group.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+		log.Debug("application.AddGroup> proj=%s app=%s group=%s", proj.Name, a.Name, g.Name)
+		groupAttachedToApp, erra := group.CheckGroupInApplication(db, a.ID, g.ID)
+		if erra != nil {
+			log.Warning("AddGroup> Unable to check group in application %s", erra)
+			return erra
+		}
+
+		if !groupAttachedToApp {
+			if err := group.InsertGroupInApplication(db, a.ID, g.ID, gp.Permission); err != nil {
+				log.Warning("AddGroup> Unable to inserting in application_group %d %d %d: s%", a.ID, g.ID, gp.Permission, err)
```
Comment (sguiheux) [inline engine/api/application/application_group.go]: s% => %s

File: tests/sc_artifact.yml [hunk 3]
```diff
@@ -15,7 +16,7 @@ testcases:
   - script: cds action add step IT_SC_cdsartifacttest "Artifact Upload" -p path=./logs.txt -p tag="{{.cds.version}}"
   - script: cds pipeline job add ITSCARTIFACT ArtifactPip myJob1
   - script: cds pipeline job append ITSCARTIFACT ArtifactPip myJob1 IT_SC_cdsartifacttest
-  - script: cds pipeline run ITSCARTIFACT TestApp ArtifactPip
+  - script: cds pipeline run ITSCARTIFACT TestApp ArtifactPip TestEnv
```
Comment (sguiheux) [inline tests/sc_artifact.yml]: ArtifactPip must be a deployment pipeline

Comment (fsamin) [inline tests/sc_artifact.yml]: fixed

**Pull request merged**